docs/topics/proc_debug-logging-nodejs.adoc

```diff
@@ -0,0 +1,162 @@
+[[debug-logging]]
+= Runtime Application Debug Logging
+
+Debug logging is a way to add granular logging to an application that is explicitly enabled when performing debugging. This allows you to keep this logging within your application and only enable it when needed.
+
+
+== Add Debug Logging
+There are several options for debug logging, but this example uses the link:https://www.npmjs.com/package/debug[debug package].
+
+[discrete]
+=== Prerequisites 
+* An application you want to debug. For example, you can use xref:mission-http-api-nodejs[a booster].
+
+[discrete]
+=== Procedure
+
+. Add the `debug` logging definition.
++
+[source,js,options="nowrap",subs="attributes+"]
+----
+var debug = require('debug')('mybooster');
+----
+
+. Add debug statements.
++
+[source,js,options="nowrap",subs="attributes+"]
+----
+app.use('/api/greeting', (request, response) => {
+  const name = request.query ? request.query.name : undefined;
+  //log name in debugging
+  debug('name: '+name);
+  response.send({content: `Hello, ${name || 'World'}`});
+});
+----
+
+== Show Debug Logging on Localhost
+
+Use the the `DEBUG` environment variable when starting your application to enable debug logging.
+
+[discrete]
+=== Prerequisites 
+* An application with debug logging.
+
+[discrete]
+=== Procedure
+
+. Start your application with the `DEBUG` environment variable set to enable debug logging.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ DEBUG=mybooster npm start
+----
++
+NOTE: The `DEBUG` environment variable uses link:https://www.npmjs.com/package/debug#wildcards[wildcards] to filter debugging messages.
+
+. Test your application to invoke debug logging. 
++
+For example, if you had debug logging in the xref:mission-http-api-nodejs[{name-mission-http-api} booster] to log the `name` variable in the `/api/greeting` method:
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ curl http://localhost:8080/api/greeting?name=Sarah
+----
+
+. View your application logs to see your debug messages.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+mybooster name: Sarah +3m
+----
+
+
+== Show Debug Logging on OpenShift
+
+Use the the `DEBUG` environment variable in your application pod in OpenShift to enable debug logging.
+
+[discrete]
+=== Prerequisites 
+* An application with debug logging.
+* The `oc` CLI client installed.
+
+
+[discrete]
+=== Procedure
+
+. Use the `oc` CLI client to log into your OpenShift instance.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ oc login ...
+----
+
+. Deploy your application to OpenShift.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ npm run openshift
+----
++
+This runs the `openshift` `npm` script, which wraps direct calls to xref:about-nodeshift[nodeshift].
```

Comment:
I can't quite make out the meaning of the sentence when both _openshift_ and _npm_ are verbatim. Is it a script executed by npm? If so, I believe npm shouldn't be verbatim because it's just a name of a program, not its invocation. If not, some other formatting would be in order IMO.

Reply:
fixed